doc/buildAndProgram.md

```diff
@@ -35,7 +35,8 @@ CMake configures the project according to variables you specify the command line
 **GDB_CLIENT_BIN_PATH**|Path to arm-none-eabi-gdb executable. Used only if `USE_GDB_CLIENT` is 1.|`-DGDB_CLIENT_BIN_PATH=/home/jf/nrf52/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gdb`
 **GDB_CLIENT_TARGET_REMOTE**|Target remote connection string. Used only if `USE_GDB_CLIENT` is 1.|`-DGDB_CLIENT_TARGET_REMOTE=/dev/ttyACM0`
 **BUILD_DFU (\*\*)**|Build DFU files while building (needs [adafruit-nrfutil](https://github.com/adafruit/Adafruit_nRF52_nrfutil)).|`-DBUILD_DFU=1`
-**WATCH_COLMI_P8**|Use pin configuration for Colmi P8 watch|`-DWATCH_COLMI_P8=1`
+**TARGET_DEVICE**|Target device, used for pin map. Allowed: `PINETIME, P8`|`-DTARGET_DEVICE=PINETIME` (Default)
+**LF_CLK**|Configures the LF clock source. Allowed: `RC, XTAL, SYNT`|`-DLF_CLK=XTAL` (Default)
```

Comment:
As I wrote in https://github.com/InfiniTimeOrg/InfiniTime/pull/1051#discussion_r862996876, I think the hardware configuration should be based on the target device. What I mean by this, is that enabling PineTime target device should automatically enable XTAL for example. Having `LF_CLK` for example is good abstraction, but my idea is that they don't need to be directly controllable by the user/developer.

Reply:
This abstraction could be solved entirely in CMake, which would then set the required "fine-grained" configuration definitions. Alternatively, I could rewrite the various drivers etc. to depend only on a tingle target selection parameter. What would you suggest?

Reply:
The first option sounds good.